src/PlusServer/Commands/vtkPlusCapistranoCommand.cxx

```diff
@@ -0,0 +1,339 @@
+  }
+
+  std::string DeviceId = (aDevice->GetDeviceId().empty() ? "(unknown)" : aDevice->GetDeviceId());
+  std::string resultString = "<CommandReply>";
+  std::string error = "";
+  std::map < std::string, std::pair<IANA_ENCODING_TYPE, std::string> > metaData;
+  PlusStatus status = PLUS_SUCCESS;
+
+  std::list<std::pair<std::string, std::string>>::iterator paramIt;
+  for (paramIt = this->RequestedParameterChanges.begin(); paramIt != this->RequestedParameterChanges.end(); ++paramIt)
+  {
+    std::string parameterName = paramIt->first;
+    std::string value = paramIt->second;
+    resultString += "<Parameter Name=\"" + parameterName + "\"";
+
+    if (parameterName != vtkPlusCapistranoVideoSource::CAPISTRANO_SET_WOBBLE_RATE
+        && parameterName != vtkPlusCapistranoVideoSource::CAPISTRANO_GET_WOBBLE_RATE
+        && parameterName != vtkPlusCapistranoVideoSource::CAPISTRANO_SET_MIS_MODE
+        && parameterName != vtkPlusCapistranoVideoSource::CAPISTRANO_SET_MIS_PULSE_PERIOD
+        && parameterName != vtkPlusCapistranoVideoSource::CAPISTRANO_GET_JITTER_COMPENSATION
+        && parameterName != vtkPlusCapistranoVideoSource::CAPISTRANO_SET_JITTER_COMPENSATION
+        && parameterName != vtkPlusCapistranoVideoSource::CAPISTRANO_FREEZE_PROBE
+        && parameterName != vtkPlusCapistranoVideoSource::CAPISTRANO_IS_PROBE_FROZEN
+        && parameterName != vtkPlusCapistranoVideoSource::CAPISTRANO_SET_BIDIRECTIONAL_MODE)
+    {
+      error += "Invalid parameter " + parameterName + ". ";
+      resultString += " Success=\"false\"/>";
+      metaData[parameterName] = std::make_pair(IANA_TYPE_US_ASCII, "FAIL");
+      status = PLUS_FAIL;
+      continue;
+    } 
+    vtkPlusCapistranoVideoSource* device = dynamic_cast<vtkPlusCapistranoVideoSource*>(aDevice);
+
+    if (parameterName == vtkPlusCapistranoVideoSource::CAPISTRANO_SET_WOBBLE_RATE)
+    {
+      int wobbleRate_int = std::stoi(value);
+      status = device->SetWobbleRate((unsigned char)wobbleRate_int);
+      resultString += " Success=\"true\"/>";
+      metaData[parameterName] = std::make_pair(IANA_TYPE_US_ASCII, "SUCCESS");
+    }
+    if (parameterName == vtkPlusCapistranoVideoSource::CAPISTRANO_GET_WOBBLE_RATE)
+    {
+      value = std::to_string(device->GetWobbleRate());
+      std::stringstream ss;
+      ss << value;
+      resultString += " Success=\"true\"";
+      resultString += " Value=\"" + ss.str() + "\"";
+      metaData[parameterName] = std::make_pair(IANA_TYPE_US_ASCII, ss.str());
+      status = PLUS_SUCCESS;
+    }
+    if (parameterName == vtkPlusCapistranoVideoSource::CAPISTRANO_SET_BIDIRECTIONAL_MODE)
+    {
+      bool bidi = std::stoi(value);
+      status = device->SetBidirectionalMode(bidi);
+      resultString += " Success=\"true\"/>";
+      metaData[parameterName] = std::make_pair(IANA_TYPE_US_ASCII, "SUCCESS");
+    }
+    if (parameterName == vtkPlusCapistranoVideoSource::CAPISTRANO_FREEZE_PROBE)
+    {
+      bool freezeProbe_bool  = std::stoi(value);
```

Comment:
it's a little weird that the setter for the frozen parameter takes as input ("0"/"1") while the getter returns ("true"/"false"). we should probably be consistent here?

additionally, if you try to send a string like "true" through this command with these changes, the server does crash. do you think it's necessary to return parse failures like in https://github.com/PlusToolkit/PlusLib/blob/9735b2b70e7db9996dd7fb8534e85b58fb90985c/src/PlusServer/Commands/vtkPlusSetUsParameterCommand.cxx#L220 ?

Reply:
Is there a preference for using true/false vs 0/1? Using 0/1 makes parsing arguments easier - like the UsParameters example. True/False may be more intuitive for a user, but the returns will most often be handled by SonoEQ, so I am not sure how important that is

Reply:
I think 0/1 makes sense. @jamesobutler do you have a preference here?

Reply:
Sorry didn't see the edit here. Carry on

Reply:
> I think 0/1 makes sense. @jamesobutler do you have a preference here?

I would agree, I feel like the code is cleaner with 0/1, but I then wasn't sure about continuity